app/lib/actions/remote_execution/run_hosts_job.rb

```diff
@@ -25,15 +27,27 @@ def create_sub_plans
         job_invocation = JobInvocation.find(input[:job_invocation_id])
         proxy_selector = RemoteExecutionProxySelector.new
 
-        # composer creates just "pattern" for template_invocations because target is evaluated
-        # during actual run (here) so we build template invocations from these patterns
-        job_invocation.targeting.hosts.map do |host|
+        current_batch.map do |host|
+          # composer creates just "pattern" for template_invocations because target is evaluated
+          # during actual run (here) so we build template invocations from these patterns
           template_invocation = job_invocation.pattern_template_invocation_for_host(host).deep_clone
           template_invocation.host_id = host.id
           trigger(RunHostJob, job_invocation, host, template_invocation, proxy_selector)
         end
       end
 
+      def batch(from, size)
+        hosts.slice(from, size)
```

Comment:
Would it be possible to prevent running the full sql on the whole set, and use limit sql capabilities for this? From what I've seen, the `slice` is first evaluating the arel and then just slicing the original result, which might be quite expensive operation.